andela/wg-kalii
# #154953745  add support for macro, meso and microcycles

#### What does this PR do?

Add support for macro, meso and microcycles

#### Description of Task to be completed?

A macrocycle is a year-long plan, a mesocycle is planning for 2 - 6 weeks and a micro is a week-long training cycle.

These cycles are centered around creating a training cycle that stretches over a weeklong period.
#### How should this be manually tested?

1. A user when logged in needs to first create a workout that suits them.
2. after this has been created the user can then select workout scheduling and ensure that periodization is selected.
3. once selected they can choose the duration of their periodization

#### Any background context you want to provide?

Before this workouts could only be selected for a period spanning 1 - 25 weeks

#### What are the relevant pivotal tracker stories?

#154953745
#### Screenshots (if appropriate)
Dashboard view
<img width="1436" alt="screen shot 2018-03-16 at 02 44 28" src="https://user-images.githubusercontent.com/30991429/37496591-77acae2e-28c4-11e8-8af3-566f123b1714.png">

Creating workout
<img width="1437" alt="screen shot 2018-03-16 at 02 45 18" src="https://user-images.githubusercontent.com/30991429/37496622-a69165ea-28c4-11e8-9cee-83c0d0a73c2c.png">

Workout Created
<img width="1437" alt="screen shot 2018-03-16 at 02 46 12" src="https://user-images.githubusercontent.com/30991429/37496637-b4c52854-28c4-11e8-8874-060d01707968.png">

Current scedule list
<img width="1437" alt="screen shot 2018-03-16 at 02 46 35" src="https://user-images.githubusercontent.com/30991429/37496644-c103fda2-28c4-11e8-85ef-159e24ecbfec.png">

Creating Schedule
<img width="595" alt="screen shot 2018-03-16 at 02 47 00" src="https://user-images.githubusercontent.com/30991429/37496657-d7b436ca-28c4-11e8-85ad-d623c2254f94.png">

Blank Schedule
<img width="1438" alt="screen shot 2018-03-16 at 02 47 10" src="https://user-images.githubusercontent.com/30991429/37496668-e264882c-28c4-11e8-9ae1-7445242be520.png">

Adding a workout to schedule
<img width="597" alt="screen shot 2018-03-16 at 02 47 29" src="https://user-images.githubusercontent.com/30991429/37496676-eec6ae1a-28c4-11e8-9dbe-

View of workout added to schedule
<img width="1439" alt="screen shot 2018-03-16 at 02 47 46" src="https://user-images.githubusercontent.com/30991429/37496682-fc268f62-28c4-11e8-91c4-07b36f547c6d.png">
79ea7408beb7.png">
